### PR TITLE
Fixing tnull value bug in column insert. Closes #7555

### LINKED ISSUE #7555: Possible bug with TNULL value being changed on FITS binary table to Astropy table roundtrip
I'm pretty sure this is my last `Tables` bug **crosses fingers**.

So once again, I'm in Astropy version 3.1.dev21825.

When I round trip a FITS binary table that has some columns with a TNULL keyword set, all the TNULL keyword values and all data that had the TNULL data value in the table, get changed to `999999`.  This only happens when I add (or update) a whole row at once.  This includes my final test that I'll post below where I add a new row, remove the new row, and save out the file.

It does NOT happen when I update a single value, change any of the masking, or add a new column.

Here's the code used to do the roundtrip and the fitsdiff output from the original vs the round tripped file:

```
def round_trip(filename, ext=1):
    with fits.open(filename) as fits_file:
        tab = Table.read(filename, hdu=ext)
        tab.add_row(tab[-1])
        tab = tab[:-1]
        new_tab = fits.table_to_hdu(tab)
        fits_file[ext] = new_tab
        fits_file.writeto(filename.replace(".fits","_rt.fits"), 
                          overwrite=True)
```

Fitsdiff:
screen shot 2018-06-11 at 11 43 28 am

screen shot 2018-06-11 at 11 43 40 am

screen shot 2018-06-11 at 11 44 01 am

### PR DESCRIPTION
**Pull request by SaOgaz**
All tests passed for me locally, and this seems to have fixed the problem for my personal test files!

EDIT: Closes #7555

### CODE REVIEW

**Comment by SaOgaz** [comment]
Do bug fixes need a changelog entry?

**Comment by saimn** [comment]
@SaOgaz - Yes, needs a changelog entry. I have set the milestone to 2.0.x as I guess it should be backported, but it's up to @taldcroft or @mhvk.

**Comment by mhvk** [comment]
Looks good to me. I think a backport makes sense, but have lost track slightly. @taldcroft?

**Comment by taldcroft** [comment]
This definitely looks like it could be backported as it is quite narrow in impact code wise.

@saogaz - what about #4707?  Were they any other issues with fill value getting dropped?

**Comment by SaOgaz** [comment]
@taldcroft I don't know about that issue specifically, I wasn't using the filtering and after reading through it I'm not confident these two issues are related.  Filtering doesn't call insert does it?

**Comment by taldcroft** [comment]
Correct, slicing (aka filtering) does not use insert.  So ok with staying in this narrow scope here.

File: CHANGES.rst [hunk 1]
```diff
@@ -1816,6 +1816,10 @@ astropy.table
   ``Time``, ``SkyCoord``, ``Latitude``, and ``EarthLocation`` via the
   ASCII ECSV format. [#6181]
 
+- Bug fix for ``MaskedColumn`` insert method, where ``fill_value`` attribute
+  was not being passed along to the copy of the ``MaskedColumn`` that was
+  returned. [#7555]
```
Comment (bsipocz) [inline CHANGES.rst]: Please add the PR number here rather than the issue number.

**Comment by SaOgaz** [comment]
Can someone leave an approved review on this?  Looks like everyone is okay with it...

**Pull request merged**